### PR TITLE
Add build-microk8s-charm job definition

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Add Jenkins job for building MicroK8s charm and uploading to CharmHub

### CODE REVIEW

**Comment by neoaggelos** [comment]
cc @ktsakalozos

**Comment by ktsakalozos** [comment]
LGTM +1

File: jobs/build-microk8s-charm.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      Build MicroK8s charm and upload revision to CharmHub.
+    project-type: freestyle
+    node: runner-amd64
+    wrappers:
+      - ansicolor
+      - timestamps
+      - workspace-cleanup
+      - credentials-binding:
+          - file:
+              credential-id: JUJUCONTROLLERS_SERVERSTACK
```
Comment (ktsakalozos) [inline jobs/build-microk8s-charm.yaml:15]: @johnsca @kwmonroe this job uses our internal serverstack. Do you see any issue with that?

Comment (kwmonroe) [reply jobs/build-microk8s-charm.yaml:15]: Only concern would be left-over resources, but it looks like the `cleanup` code in this PR should tear everything down nicely.

**Review by kwmonroe — APPROVED** [review]

**Pull request merged**